presto-main/src/main/java/io/prestosql/operator/scalar/BitwiseFunctions.java

```diff
@@ -75,4 +85,169 @@ public static long bitwiseXor(@SqlType(StandardTypes.BIGINT) long left, @SqlType
     {
         return left ^ right;
     }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.TINYINT)
+    public static long bitwiseLeftShiftTinyint(@SqlType(StandardTypes.TINYINT) long value, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        if (shift >= 64) {
+            return 0L;
+        }
+        long shifted = (value << shift);
+        return preserveSign(shifted, TINYINT_MASK, TINYINT_SIGNED_BIT);
+    }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.SMALLINT)
+    public static long bitwiseLeftShiftSmallint(@SqlType(StandardTypes.SMALLINT) long value, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        if (shift >= 64) {
+            return 0L;
+        }
+        long shifted = (value << shift);
+        return preserveSign(shifted, SMALLINT_MASK, SMALLINT_SIGNED_BIT);
+    }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.INTEGER)
+    public static long bitwiseLeftShiftInteger(@SqlType(StandardTypes.INTEGER) long value, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        if (shift >= 64) {
+            return 0L;
+        }
+        long shifted = (value << shift);
+        return preserveSign(shifted, INTEGER_MASK, INTEGER_SIGNED_BIT);
+    }
+
+    @Description("bitwise left shift")
+    @ScalarFunction("bitwise_left_shift")
+    @SqlType(StandardTypes.BIGINT)
+    public static long bitwiseLeftShiftBigint(@SqlType(StandardTypes.BIGINT) long value, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        if (shift >= 64) {
+            return 0L;
+        }
+        return value << shift;
+    }
+
+    private static long preserveSign(long shiftedValue, long mask, long signedBit)
+    {
+        if ((shiftedValue & signedBit) != 0) {
+            // Preserve the sign in 2's complement format
+            return shiftedValue | ~mask;
+        }
+
+        return shiftedValue & mask;
+    }
+
+    @Description("bitwise logical right shift")
+    @ScalarFunction("bitwise_right_shift")
+    @SqlType(StandardTypes.TINYINT)
+    public static long bitwiseRightShiftTinyint(@SqlType(StandardTypes.TINYINT) long value, @SqlType(StandardTypes.INTEGER) long shift)
+    {
+        if (shift >= 64) {
+            return 0L;
+        }
+        return (value & TINYINT_MASK) >>> shift;
```

Comment:
This mask will turn a negative tinyint into a positive one. For example, this will fail:

```
bitwise_right_shift(tinyint '-128', 0)
```